pkg/placement/placement.go

```diff
@@ -266,12 +268,12 @@ func (p *Service) ReportDaprStatus(stream placementv1pb.Placement_ReportDaprStat
 
 	clientID, err := p.validateClient(stream)
 	if err != nil {
-		return err
+		return apiErrors.PlacementServiceInternalError(fmt.Sprintf("internal or unauthenticated error: %s", err))
```

Comment:
We can't replace the error `validateClient` returns here, because it could be different error types. Please take a closer look at the function and update the errors in there directly.